File: solutions/clojure/raindrops/1/src/raindrops.clj
(ns raindrops)

(defn convert [num]
  (let [pling-plang-plong (str (when (zero? (mod num 3))
                                 "Pling")
                               (when (zero? (mod num 5))
                                 "Plang")
                               (when (zero? (mod num 7))
                                 "Plong"))]
    (if (= pling-plang-plong "")
      (str num)
      pling-plang-plong)))

File: solutions/ocaml/raindrops/1/raindrops.ml
let raindrop n = 
  n 
  |> (fun n -> if n mod 3 = 0 then "Pling" else "") 
  |> (fun s -> if n mod 5 = 0 then s ^ "Plang" else s) 
  |> (fun s -> if n mod 7 = 0 then s ^ "Plong" else s)
  |> (fun s -> if s = "" then string_of_int n else s)